Improve oxford_iiit_pet dataset

enricoschroeder opened the pull request:
Improve oxford_iiit_pet dataset:
- Include head bounding boxes for the training split.
- Exclude samples with invalid or corrupt image files (preventing 'Corrupt JPEG data:...' error messages while using the dataset).
- Use up-to-date download URL (from official website https://www.robots.ox.ac.uk/~vgg/data/pets/))

Thank you for your contribution!

Please read https://www.tensorflow.org/datasets/contribute#pr_checklist to make sure your PR follows the guidelines.

The checklist does not apply since this is only a minor fix to an existing dataset.

fineguy commented on tensorflow_datasets/datasets/oxford_iiit_pet/oxford_iiit_pet_dataset_builder.py:
```diff
@@ -79,6 +118,7 @@ def _info(self):
             "segmentation_mask": tfds.features.Image(
                 shape=(None, None, 1), use_colormap=True
             ),
+            "head": tfds.features.BBoxFeature()
```
How about a more explicit feature name, e.g. "head_bbox"?

fineguy commented on tensorflow_datasets/datasets/oxford_iiit_pet/oxford_iiit_pet_dataset_builder.py:
```diff
@@ -126,18 +166,31 @@ def _generate_examples(
+            head_bbox = _get_head_bbox(os.path.join(xmls_dir_path, xml_name))
+        except tf.errors.NotFoundError:
+            # test samples do not have an annotation file
+            head_bbox = tfds.features.BBox(0., 0., 0., 0.)
```
Move this to a constant `_EMPTY_BBOX`.

enricoschroeder commented:
@fineguy I have addressed your comments. The corrupt samples are now being fixed instead of skipped. For that, we still need to import Tensorflow (even though data reading is now done via etils).
